src/App.tsx

```diff
@@ -1,14 +1,50 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filterBy, setFilterBy] = useState<Filter>(Filter.ALL);
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+      })
+      .finally(() => setLoading(false));
+  }, []);
+
+  const preparedTodos = useMemo(() => {
+    return todos
+      .filter((todo: Todo) => {
+        switch (filterBy) {
+          case Filter.ACTIVE:
+            return !todo.completed;
+
+          case Filter.COMPLETED:
+            return todo.completed;
+
+          default:
+            return todo;
+        }
+      })
+      .filter(todo =>
+        todo.title.toLowerCase().trim().includes(query.trim().toLowerCase()),
```

Comment:
Could you do it in one filter? And probably if the filter is `Filter.ALL` and you don't have anything in query you don't need to filter at all